Test flake: deploy_test_triggers can fail with deploymentConfig update conflicts

Observed in https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/6614/console for #5530.

```
--- FAIL: TestTriggers_manual-2 (0.05s)
	deploy_trigger_test.go:304: Starting test openshift
	deploy_trigger_test.go:91: Couldn't create updated DeploymentConfig: deploymentConfig "config" cannot be updated: the object has been modified; please apply your changes to the latest version and try again &api.DeploymentConfig{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"config", GenerateName:"", Namespace:"integration", SelfLink:"/oapi/v1/namespaces/integration/generatedeploymentconfigs/config", UID:"4fe7be5f-81ac-11e5-854d-0e93450de951", ResourceVersion:"10754", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:63582098077, nsec:0, loc:(*time.Location)(0x467cda0)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, Triggers:[]api.DeploymentTriggerPolicy{api.DeploymentTriggerPolicy{Type:"Manual", ImageChangeParams:(*api.DeploymentTriggerImageChangeParams)(nil)}}, Template:api.DeploymentTemplate{Strategy:api.DeploymentStrategy{Type:"Recreate", CustomParams:(*api.CustomDeploymentStrategyParams)(nil), RecreateParams:(*api.RecreateDeploymentStrategyParams)(nil), RollingParams:(*api.RollingDeploymentStrategyParams)(nil), Resources:api.ResourceRequirements{Limits:api.ResourceList{"cpu":resource.Quantity{Amount:10.000, Format:"DecimalSI"}, "memory":resource.Quantity{Amount:10000000000.000, Format:"DecimalSI"}}, Requests:api.ResourceList(nil)}}, ControllerTemplate:api.ReplicationControllerSpec{Replicas:1, Selector:map[string]string{"a":"b"}, Template:(*api.PodTemplateSpec)(0xc208a1d8c0)}}, LatestVersion:1, Details:(*api.DeploymentDetails)(0xc208aad2c0)}
```

Probably another fight with the deploymentConfig controller updating the  status message.

--- a/test/integration/deploy_trigger_test.go
+++ b/test/integration/deploy_trigger_test.go
@@ -10,8 +10,6 @@ import (
 	"sync"
 	"testing"
 
-	"github.com/golang/glog"
-
 	kapi "k8s.io/kubernetes/pkg/api"
 	klatest "k8s.io/kubernetes/pkg/api/latest"
 	"k8s.io/kubernetes/pkg/api/rest"
@@ -21,6 +19,7 @@ import (
 	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/master"
 	"k8s.io/kubernetes/pkg/tools/etcdtest"
+	"k8s.io/kubernetes/pkg/util/wait"
 	watchapi "k8s.io/kubernetes/pkg/watch"
 	"k8s.io/kubernetes/plugin/pkg/admission/admit"
 
@@ -47,6 +46,8 @@ import (
 	testserver "github.com/openshift/origin/test/util/server"
 )
 
+const maxUpdateRetries = 5
+
 func init() {
 	testutil.RequireEtcd()
 }
@@ -64,8 +65,6 @@ func TestTriggers_manual(t *testing.T) {
 		},
 	}
 
-	var err error
-
 	dc, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Create(config)
 	if err != nil {
 		t.Fatalf("Couldn't create DeploymentConfig: %v %#v", err, config)
@@ -77,21 +76,25 @@ func TestTriggers_manual(t *testing.T) {
 	}
 	defer watch.Stop()
 
-	config, err = openshift.Client.DeploymentConfigs(testutil.Namespace()).Generate(config.Name)
-	if err != nil {
-		t.Fatalf("Error generating config: %v", err)
+	retryErr := kclient.RetryOnConflict(wait.Backoff{Steps: maxUpdateRetries}, func() error {
+		config, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Generate(config.Name)
+		if err != nil {
+			return err
+		}
+		if config.LatestVersion != 1 {
+			t.Fatalf("Generated deployment should have version 1: %#v", config)
+		}
+		t.Logf("config(1): %#v", config)
+		updatedConfig, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Update(config)
+		if err != nil {
+			return err
+		}
+		t.Logf("config(2): %#v", updatedConfig)
+		return nil
+	})
+	if retryErr != nil {
+		t.Fatal(err)
 	}
-	if config.LatestVersion != 1 {
-		t.Fatalf("Generated deployment should have version 1: %#v", config)
-	}
-	glog.Infof("config(1): %#v", config)
-
-	new, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Update(config)
-	if err != nil {
-		t.Fatalf("Couldn't create updated DeploymentConfig: %v %#v", err, config)
-	}
-	glog.Infof("config(2): %#v", new)
-
 	event := <-watch.ResultChan()
 	if e, a := watchapi.Added, event.Type; e != a {
 		t.Fatalf("expected watch event type %s, got %s", e, a)
@@ -243,28 +246,35 @@ func TestTriggers_configChange(t *testing.T) {
 
 	assertEnvVarEquals("ENV1", "VAL1", deployment, t)
 
-	// submit a new config with an updated environment variable
-	if config, err = openshift.Client.DeploymentConfigs(testutil.Namespace()).Generate(config.Name); err != nil {
-		t.Fatalf("Error generating config: %v", err)
-	}
+	retryErr := kclient.RetryOnConflict(wait.Backoff{Steps: maxUpdateRetries}, func() error {
+		// submit a new config with an updated environment variable
+		config, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Generate(config.Name)
+		if err != nil {
+			return err
+		}
 
-	config.Template.ControllerTemplate.Template.Spec.Containers[0].Env[0].Value = "UPDATED"
+		config.Template.ControllerTemplate.Template.Spec.Containers[0].Env[0].Value = "UPDATED"
 
-	// before we update the config, we need to update the state of the existing deployment
-	// this is required to be done manually since the deployment and deployer pod controllers are not run in this test
-	deployment.Annotations[deployapi.DeploymentStatusAnnotation] = string(deployapi.DeploymentStatusComplete)
-	// update the deployment
-	if _, err = openshift.KubeClient.ReplicationControllers(testutil.Namespace()).Update(deployment); err != nil {
-		t.Fatalf("Error updating existing deployment: %v", err)
-	}
+		// before we update the config, we need to update the state of the existing deployment
+		// this is required to be done manually since the deployment and deployer pod controllers are not run in this test
+		deployment.Annotations[deployapi.DeploymentStatusAnnotation] = string(deployapi.DeploymentStatusComplete)
+		// update the deployment
+		if _, err := openshift.KubeClient.ReplicationControllers(testutil.Namespace()).Update(deployment); err != nil {
+			return err
+		}
 
-	event = <-watch.ResultChan()
-	if e, a := watchapi.Modified, event.Type; e != a {
-		t.Fatalf("expected watch event type %s, got %s", e, a)
-	}
+		event = <-watch.ResultChan()
+		if e, a := watchapi.Modified, event.Type; e != a {
+			t.Fatalf("expected watch event type %s, got %s", e, a)
+		}
 
-	if _, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Update(config); err != nil {
-		t.Fatalf("Couldn't create updated DeploymentConfig: %v", err)
+		if _, err := openshift.Client.DeploymentConfigs(testutil.Namespace()).Update(config); err != nil {
+			return err
+		}
+		return nil
+	})
+	if retryErr != nil {
+		t.Fatal(retryErr)
 	}
 
 	event = <-watch.ResultChan()

